zipline/data/us_equity_pricing.py

```diff
@@ -127,8 +138,14 @@ def winsorise_uint32(df, column, *columns):
         ``df`` with rows that contain values that do not fit into a uint32
         zeroed out.
     """
-    columns = (column,) + columns
-    df[reduce(op.or_, (df[c] > UINT32_MAX for c in columns))] = 0
+    mask = (df[list(columns)] > UINT32_MAX).any(axis=1)
+    if mask.any():
+        if invalid_data_behavior == 'raise':
+            raise ValueError('Values out of bounds for uint32: %r' % df[mask])
+        if invalid_data_behavior == 'warn':
+            warnings.warn('Values out of bounds for uint32: %r' % df[mask])
```

Comment:
This might want to explicitly say that we're ignoring those values.  Also, it'd be nice to include the count of bad values here (`mask.sum()` should work for this).